src/github/github.controller.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  Controller,
+  Query,
+  Get,
+  HttpStatus,
+  Post,
+  UseGuards,
+  HttpException,
+  Req,
+  Put,
+  Param,
+  Body,
+  Delete,
+} from '@nestjs/common';
+import { GithubService } from './github.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { Env } from 'src/env';
+import { JwtAuthGuard } from 'src/auth/jwt-auth.guard';
+import { UpdatePracticesDto } from './dto/update.practices.dto';
+import { TRequest } from 'src/common/types/user.types';
+import { GetApiQueryParamsSwaggerWithOrganizationName } from 'src/decorator/decorator';
+
+@Controller('github')
+@ApiTags('Github')
+export class GithubController {
+  constructor(private readonly githubService: GithubService) {}
+
```

Comment:
PullProofAI

Consider extracting the logic in the controller methods into separate service methods for better separation of concerns.
```typescript
// Example of extracting logic into service methods:

// GithubController
async listRepos(
  @Req() req: TRequest,
  @Query('organizationName') organizationName: string,
  @Query('page') page: number = 1,
  @Query('limit') limit: number = 20,
) {
  try {
    return await this.githubService.listRepos(req, organizationName, page, limit);
  } catch (error) {
    this.handleError(error);
  }
}

private handleError(error: any) {
  if (error.status) {
    throw new HttpException(error.message, error.status);
  } else {
    throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
  }
}

// Repeat the pattern for other controller methods
```
By moving the business logic to service methods, the controller becomes leaner and adheres more closely to the single responsibility principle.


<!-- This is an auto-generated comment by PullProofAI -->